test/integration/container_volume_caching_test.js

```diff
@@ -354,4 +360,59 @@ suite('volume cache tests', function () {
     );
     yield worker.terminate();
   }));
+
+  test('purge cache after run task', co(function* () {
+    var cacheName = 'tmp-obj-dir-' + Date.now().toString();
+    var neededScope = 'docker-worker:cache:' + cacheName;
+    var fullCacheDir = path.join(localCacheDir, cacheName);
+    settings.configure({
+      cache: {
+        volumeCachePath: volumeCacheDir
+      },
+      garbageCollection: {
+        interval: 100
+      }
+    });
+
+    var task = {
+      payload: {
+        image: 'taskcluster/test-ubuntu',
+        command: cmd(
+          'echo "foo" > /tmp-obj-dir/foo.txt',
+          'ls /tmp-obj-dir'
+        ),
+        features: {
+          localLiveLog: true
+        },
+        cache: {},
+        maxRunTime:         5 * 60
+      },
+      scopes: [neededScope]
+    };
+
+    task.payload.cache[cacheName] = '/tmp-obj-dir';
+
+    let worker = new TestWorker(DockerWorker);
+    yield worker.launch();
+    let result = yield worker.postToQueue(task);
+
+    assert.equal(result.run.state, 'completed');
+    assert.equal(result.run.reasonResolved, 'completed');
+
+    var purgeCache = new taskcluster.PurgeCache({
+      credentials: {
+        clientId: process.env.TASKCLUSTER_CLIENT_ID,
+        accessToken: process.env.TASKCLUSTER_ACCESS_TOKEN
+      }
+    });
+
+    purgeCache.purgeCache(worker.provisionerId, worker.workerType, {
+      cacheName: cacheName
+    });
+
+    yield sleep(2000);
```

Comment:
so I think there are ways around introducing sleeps here.  Basically I think what you're trying to wait for is that the removal actually happened.  There are times where it might take longer than 2 seconds for a pulse message to be sent, received by the worker, and the directory removed.